### DISCUSSION TITLE
A question about add resource bundle

### DISCUSSION THREAD

**Discussion by Rednegniw** [i18next/next-i18next#1669] [Q&A] [answered]
Hi. It seems I can't figure out your example about adding a namespace client side on page load. I don't want to use `getServerSideProps` since most of my app is a client side app that dynamically loads data with `swr`. It also has some page transition animations that would be broken if I used that feature.

![image](https://user-images.githubusercontent.com/16265289/152639746-b8d2327f-321d-44c0-8969-88e27d076f8b.png)

There are two problems:
- i18n is `null` at first, but it seems it doesn't initialize automatically.
- the `addResourceBundle` function doesn't actually only take two arguments.. at least TypeScript doesn't think so.

How could I go about solving this?

**Comment by isaachinman** [comment, marked as answer]
> i18n is null at first, but it seems it doesn't initialize automatically.

Please see a reply to that matter [here](https://github.com/isaachinman/next-i18next/issues/918#issuecomment-1032754263).

> the addResourceBundle function doesn't actually only take two arguments.. at least TypeScript doesn't think so.

The documentation for that function is [here](https://www.i18next.com/overview/api#addresourcebundle). You can see there are actually five arguments in total. If the types are incorrect, it would be great to open a PR against the i18next repo. cc @adrai.
